ci/do_ci.sh

```diff
@@ -245,21 +245,13 @@ elif [[ "$1" == "bazel.coverage" ]]; then
   # after 0.21.
   [ -z "$CIRCLECI" ] || export BAZEL_TEST_OPTIONS="${BAZEL_TEST_OPTIONS} --local_resources=12288,4,1"
 
-  # There is a bug in gcovr 3.3, where it takes the -r path,
-  # in our case /source, and does a regex replacement of various
-  # source file paths during HTML generation. It attempts to strip
-  # out the prefix (e.g. /source), but because it doesn't do a match
-  # and only strip at the start of the string, it removes /source from
-  # the middle of the string, corrupting the path. The workaround is
-  # to point -r in the gcovr invocation in run_envoy_bazel_coverage.sh at
-  # some Bazel created symlinks to the source directory in its output
-  # directory. Wow.
-  cd "${ENVOY_BUILD_DIR}"
-  SRCDIR="${GCOVR_DIR}" "${ENVOY_SRCDIR}"/test/run_envoy_bazel_coverage.sh
+  test/run_envoy_bazel_coverage.sh
   collect_build_profile coverage
   exit 0
 elif [[ "$1" == "bazel.clang_tidy" ]]; then
   setup_clang_toolchain
+  # TODO(wu-bin): Remove once https://github.com/envoyproxy/envoy/pull/6229 is merged.
+  export BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} --linkopt='-Wl,--allow-multiple-definition'"
```

Comment:
I don't get why this is needed and how #6229 makes it better.

Reply:
This is also a hack for the "multiple definition" errors I mentioned above. @lizan said we probably don't want this flag(--allow-multiple-definition) for bazel.release, but for other ci this flag is fine.

Reply:
Ack, resolving, let's continue discussion on #6229 above.